packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -369,19 +381,74 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
         // are in the same order.
         this.usedRoutes = usedRoutes.sort();
 
-        // If this node is referenced, clear unreferencedTimestamp, if any.
-        // If this node is not referenced and unreferencedTimestamp is undefined, it just became unreferenced. Update
-        // unreferencedTimestamp to the gcTimestamp.
+        // If GC is not disabled and we are tracking a summary, update the work-in-progress used routes so that it can
+        // be tracked for this summary.
+        if (!this.gcDisabled && this.isTrackingInProgress()) {
+            this.wipSerializedUsedRoutes = JSON.stringify(this.usedRoutes);
+        }
+
         if (this.isReferenced()) {
+            // If this node has been unreferenced for longer than maxUnreferencedTimeMs and is being referenced,
+            // log an error as this may mean the maxUnreferencedTimeMs is not long enough.
+            this.logErrorIfExpired("expiredObjectRevived", gcTimestamp);
+
+            // Clear unreferenced / expired state, if any.
+            this.expired = false;
             this.unreferencedTimestamp = undefined;
-        } else if (this.unreferencedTimestamp === undefined) {
+            this.unreferencedTimer.clear();
+            return;
+        }
+
+        // This node is unreferenced. We need to check if this node has expired or if we need to start the unreferenced
+        // timer which would mark the node as expired.
+
+        // If there is no timestamp when GC was run, we don't have enough information to determine whether this content
+        // should be expired.
```

Comment:
Under what conditions would this happen?

Reply:
The `gcTimestamp` that is passed here is the timestamp of the last op processed by this client. If there are no ops in the system when GC is run, this will be undefined. For example, an on-demand summary can be requested when there are on ops or when we have pluggable heuristics, it can decide to summarize in the absence of ops.

Maybe we should not allow summaries in absence of ops since nothing changed and just return the previous summary but we don't have that yet.